week1/1.5/Notes/Notes/Controllers/Model Controllers/NoteController.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  NoteController.swift
+//  Notes
+//
+//  Created by Eric Lanza on 7/6/18.
+//  Copyright © 2018 ETLanza. All rights reserved.
+//
+
+import Foundation
+
+class NoteController: Codable {
+    
+    static let shared = NoteController()
+    
+    var notes = [Note]()
+    
+    //MARK: - CRUD Methods
+    func addNote(_ note: Note) {
+        notes.append(note)
+        saveToPersistentStore()
+    }
+    
+    func updateNote(_ note: Note, withBodyText bodyText: String) {
+        note.bodyText = bodyText
+        saveToPersistentStore()
+    }
+    
+    func deleteNote(_ note: Note) {
+        guard let indexOfNote = NoteController.shared.notes.index(of: note) else { return }
+        notes.remove(at: indexOfNote)
+        saveToPersistentStore()
+    }
+    
+    //MARK: - Persistance Methods
+    func fileURL() -> URL {
+        let paths = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)
+        let documentDirectory = paths[0]
+        let fileName = "notes.json"
+        let url = documentDirectory.appendingPathComponent(fileName)
+        return url
+    }
+    
+    func saveToPersistentStore() {
+        let jsonEncoder = JSONEncoder()
```

Comment:
static let encoder = JSONEncoder()

is a similar optimization as mentioned above.